src/auth/auth-api-request.ts

```diff
@@ -961,6 +968,31 @@ export abstract class AbstractAuthRequestHandler {
           'Properties argument must be a non-null object.',
         ),
       );
+    } else if (validator.isNonNullObject(properties.providerToLink)) {
```

Comment:
Aren't the validations at validateCreateEditRequest sufficient? Seems like the validation is added in two places.